src/inmanta/env.py

```diff
@@ -314,11 +316,189 @@ class PipUpgradeStrategy(enum.Enum):
     ONLY_IF_NEEDED = "only-if-needed"
 
 
+class PipConfig(BaseModel):
+    """
+    :param use_config_file: Indicates whether the pip configuration files have to be taken into account when installing
+        Python packages.
+    :param index_url: one pip index url for this project.
+    :param extra_index_url:  additional pip index urls for this project. This is generally only
+        recommended if all configured indexes are under full control of the end user to protect against dependency
+        confusion attacks. See the `pip install documentation <https://pip.pypa.io/en/stable/cli/pip_install/>`_ and
+        `PEP 708 (draft) <https://peps.python.org/pep-0708/>`_ for more information.
+    :param pre:  allow pre-releases when installing Python packages, i.e. pip --pre.
+        Defaults to False. When false and pip.use-system-config=true we follow the system config.
+    :param use_system_config: defaults to false.
+        When true, sets the pip's index url, extra index urls and pre according to the respective settings outlined above
+            but otherwise respect any pip environment variables and/or config in the pip config file,
+            including any extra-index-urls.
+
+        If no indexes are configured in pip.index-url/pip.extra-index-url, this option falls back to pip's default behavior,
+        meaning it uses the pip index url from the environment, the config file, or PyPi, in that order.
+
+        For development, it is recommended to set this option to false, both for portability
+        (and related compatibility with tools like pytest-inmanta-lsm) and for security
+        (dependency confusion attacks could affect users that aren't aware that inmanta installs Python packages).
+    """
+
+    index_url: Optional[str] = None
+    # Singular to be consistent with pip itself
+    extra_index_url: Sequence[str] = []
+    pre: bool = False
+    use_system_config: bool = False
+
+    def has_source(self) -> bool:
+        """Can this config get packages from anywhere?"""
+        return self.index_url or self.use_system_config
+
+    def assert_has_source(self, reason: str) -> None:
+        if not self.has_source():
+            # TODO: Is this the correct exception type here?
+            # It is consistent with was there before, but is it Correct
+            # TODO: import cycle
+            from inmanta.module import InvalidModuleException
+
+            raise InvalidModuleException(
```

Comment:
What exception should I raise here?

This is surprisingly impactful.

I don't think this is ideal. 

I could go for `env.PackageNotFound`

I could make a new one, two wrap at the appropriate places? (if so, what is its parent?)

Reply:
`PackageNotFound` sounds good to me.